### PR TITLE
add document generation to tox

### PR DESCRIPTION
**Pull request by gaborbernat**
and trigger automatically at CI (also check links in documentation to avoid having dangling pointers to the internet)

### CODE REVIEW

File: dateutil/zoneinfo/__init__.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 from dateutil.tz import tzfile as _tzfile
 
-__all__ = ["get_zonefile_instance", "gettz", "gettz_db_metadata", "rebuild"]
+__all__ = ["get_zonefile_instance", "gettz", "gettz_db_metadata"]
```
Comment (pganssle) [inline dateutil/zoneinfo/__init__.py]: I don't think we can change the public interface like this...

Comment (pganssle) [reply dateutil/zoneinfo/__init__.py]: Hm, actually `rebuild` is a separate file. Yeah, I see how this works. This is fine.

Comment (gaborbernat) [reply dateutil/zoneinfo/__init__.py]: the problem here is that this import does not work, the Sphinx could not load it, as no import is there. how is this rebuild suppose to exist?

**Comment by pganssle** [comment]
This is awesome, thanks @gaborbernat. I'll merge once CI passes.

**Comment by gaborbernat** [comment]
``tox -e docs`` will now build to ``.tox\docs_out`` the documentation (on any platform) 👍

**Pull request merged**